Apps/W1/HybridGP/app/src/Migration/Support/CheckBooks/GPCheckbookMSTR.table.al

```diff
@@ -182,20 +182,27 @@ table 40099 "GP Checkbook MSTR"
     procedure MoveStagingData()
     var
         BankAccount: Record "Bank Account";
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        MigrateInactiveCheckbooks: Boolean;
     begin
+        MigrateInactiveCheckbooks := false;
+        if GPCompanyAdditionalSettings.Get(CompanyName()) then
+            MigrateInactiveCheckbooks := GPCompanyAdditionalSettings."Migrate Inactive Checkbooks";
+
         if FindSet() then
             repeat
-                if not BankAccount.Get(CHEKBKID) then begin
-                    BankAccount.Init();
-                    BankAccount."No." := DelChr(CHEKBKID, '>', ' ');
-                    BankAccount.Name := DelChr(DSCRIPTN, '>', ' ');
-                    BankAccount."Bank Account No." := DelChr(BNKACTNM, '>', ' ');
-                    BankAccount."Last Check No." := GetLastCheckNumber(NXTCHNUM);
-                    BankAccount."Balance Last Statement" := Last_Reconciled_Balance;
-                    BankAccount."Bank Acc. Posting Group" := GetBankAccPostingGroup(ACTINDX);
-                    UpdateBankInfo(DelChr(BANKID, '>', ' '), BankAccount);
-                    BankAccount.Insert(true);
-                end;
+                if not BankAccount.Get(CHEKBKID) then
+                    if MigrateInactiveCheckbooks or not INACTIVE then begin
```

Comment:
Why don't you just filter out inactive entries if `MigrateInactiveCheckbooks` is set to false? That way you simply won't loop through them.

Reply:
Be careful here of two things:
1. You are on the Rec. You could introduce a local variable and set the filters from the rec by using copy. This way you will not change the filters that are on the Rec. 
2. Check if there is an index on this field. If the table is large it is good to add the index so the search is faster.

Reply:
I'm going to hold off on this change for now.